File: Coding Ninjas/Moderate/Reverse Linked List.py
def reverseLinkedList( head ):

    if head == None or head.next == None:
        return head
        
    curr = head.next
    head.next = None
    
    while curr != None:
        temp = curr.next
        curr.next = head
        head = curr
        curr = temp

    return head

File: Coding Ninjas/Easy/Reverse Linked List.java
public class Solution
{
    
    static Node reverseLinkedList
    ( Node head, Node tail )
    {
        if ( tail == null )
        {
            return head;
        }

        Node temp = tail.next;
        tail.next = head;
        
        return reverseLinkedList( tail, temp );
    }
    
    
    
    public static Node reverseLinkedList(Node head)
    {
        return reverseLinkedList( null, head );
    }
    
}
